### PR TITLE
ENYO-5542: UI Doc Review (A-M)

### PR DESCRIPTION
**Pull request by teckliew**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Review UI Docs A to Marquee.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Also edited a bit of Moonstone docs.

### Links
[//]: # (Related issues, references)
ENYO-5542

### Comments
Enact-DCO-1.0-Signed-off-by: Teck Liew teck.liew@lge.com

### CODE REVIEW

File: packages/ui/Button/Button.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ const ButtonBase = kind({
 		css: PropTypes.object,
 
 		/**
-		 * Applies the `disabled` class.
+		 * Disables Button and prevents events from firing.
```
Comment (webOS101) [inline packages/ui/Button/Button.js]: Is this true?

File: packages/ui/FloatingLayer/FloatingLayer.js [hunk 2]
```diff
@@ -249,12 +254,14 @@ const handleCancel = handle(
 );
 
 /**
- * {@link ui/FloatingLayer.FloatingLayer} is a component that creates an entry point to the new
- * render tree. This is used for modal components such as popups.
+ * FloatingLayer that mixes {@link ui/Cancelable.Cancelable} to handle FloatingLayer dismissal
```
Comment (webOS101) [inline packages/ui/FloatingLayer/FloatingLayer.js]: Period at end.

File: packages/ui/FloatingLayer/FloatingLayerDecorator.js [hunk 3]
```diff
@@ -1,5 +1,10 @@
 /*
- * Exports the {@link ui/FloatingLayer.FloatingLayerDecorator} higher-order component (HOC).
+ * A Higher-order Component that adds a FloatingLayer adjacent to wrapped component.
```
Comment (webOS101) [inline packages/ui/FloatingLayer/FloatingLayerDecorator.js]: Higher-order component should be lowercase, not uppercase.

File: packages/ui/Image/Image.js [hunk 4]
```diff
@@ -89,15 +89,15 @@ const ImageBase = kind({
 		css: PropTypes.object,
 
 		/**
-		 * A function that will run if the image has an error.
+		 * Called when if the image has an error.
```
Comment (webOS101) [inline packages/ui/Image/Image.js]: Can probably change this to just 'Called if'

File: packages/ui/Layout/Layout.js [hunk 5]
```diff
@@ -1,9 +1,5 @@
 /**
- * Exports the {@link ui/Layout.Layout}, {@link ui/Layout.LayoutBase}, and {@link ui/Layout.Cell}
- * components. The default export is {@link ui/Layout.Layout}. `Layout` also has two shorthand
- * components exported: `Row` and `Column`. These two assign preset `orientation` properties to
- * simplify usage and readability. They are identical to `<Layout orientation="horizontal">` and
- * `<Layout orientation="vertical">` respectively.
+ * Provides ui layout support using `Cell`, `Row`, and `Column`.
```
Comment (webOS101) [inline packages/ui/Layout/Layout.js]: Should we mention FlexBox in here anywhere?

File: packages/ui/Marquee/MarqueeBase.js [hunk 6]
```diff
@@ -58,7 +57,7 @@ const MarqueeBase = kind({
 		children: PropTypes.node,
 
 		/**
-		 * Function to capture a reference to the client node
+		 * Called to capture a reference to the client node
```
Comment (webOS101) [inline packages/ui/Marquee/MarqueeBase.js]: 'Called with a reference to the client node'?  'Called when mounting or unmounting with a ...'?  Not sure if those are better.

File: packages/ui/Marquee/MarqueeDecorator.js [hunk 7]
```diff
@@ -161,10 +160,11 @@ const MarqueeDecorator = hoc(defaultConfig, (config, Wrapped) => {
...
 
 			/**
-			 * Passed through to the wrapped component. Does not affect Marquee behavior except that
-			 * components that are `marqueeOn="focus"` will be treated as if they were
-			 * `marqueeOn="hover"`, to allow disabled (and thuse, unfocusable) components to
-			 * marquee.
+			 * Passed through to the wrapped component.
+			 *
+			 * Does not affect Marquee behavior except that components that are `marqueeOn="focus"`
+			 * will be treated as if they were `marqueeOn="hover"`, to allow disabled (and thuse,
```
Comment (webOS101) [inline packages/ui/Marquee/MarqueeDecorator.js]: typo: thus

File: packages/ui/Marquee/index.js [hunk 8]
```diff
@@ -1,6 +1,5 @@
 /**
- * A component that will marquee its overflowing contents and a higher-order component
- * (HOC) to add marquee behavior to the content of existing components.
+ * A component that will marquee its overflowing contents/.
```
Comment (webOS101) [inline packages/ui/Marquee/index.js]: slash.

File: packages/ui/FloatingLayer/FloatingLayerDecorator.js (hunk 3 above)
Comment (teckliew) [reply packages/ui/FloatingLayer/FloatingLayerDecorator.js]: Should we update this too as well?
https://github.com/enactjs/enact/blame/master/docs/contributing/documentation.md#L85

Comment (webOS101) [reply packages/ui/FloatingLayer/FloatingLayerDecorator.js]: Yes, please!

File: docs/contributing/documentation.md [hunk 9]
```diff
@@ -42,7 +42,7 @@ If possible, include an executable example showing the module using the `@exampl
 
 ## Class Level Documentation
 
-Class level documentation includes components and Higher-order Components (HOCs).  Each component or HOC export should have a class level block that discusses the features and provides an example usage. This block also serves as a place to indicate any HOCs that may be applied or to cross-reference related components.
+Class level documentation includes components and Higher-order components (HOCs).  Each component or HOC export should have a class level block that discusses the features and provides an example usage. This block also serves as a place to indicate any HOCs that may be applied or to cross-reference related components.
```
Comment (webOS101) [inline docs/contributing/documentation.md]: The H should be lowercase everywhere (except where it starts a sentence, of course).

File: packages/ui/FloatingLayer/FloatingLayerDecorator.js [hunk 10]
```diff
@@ -1,5 +1,10 @@
 /*
- * Exports the {@link ui/FloatingLayer.FloatingLayerDecorator} higher-order component (HOC).
+ * A higher-order component that adds a FloatingLayer adjacent to wrapped component.
+ *
+ * @class FloatingLayerDecorator
+ * @memberof ui/FloatingLayer.FloatingLayerDecorator
```
Comment (webOS101) [inline packages/ui/FloatingLayer/FloatingLayerDecorator.js]: This is not a jsdoc block and the @memberof is not correct.  Let's remove this change.

**Pull request merged**